pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -4278,16 +4287,33 @@ func mapNodeNameToPrivateDNSName(nodeName types.NodeName) string {
 }
 
 // mapInstanceToNodeName maps a EC2 instance to a k8s NodeName, by extracting the PrivateDNSName
-func mapInstanceToNodeName(i *ec2.Instance) types.NodeName {
+func mapInstanceToNodeName(i *ec2.Instance, explicitNames bool) types.NodeName {
+	if explicitNames {
+		for _, tag := range i.Tags {
+			key := aws.StringValue(tag.Key)
+			if key == TagNameKubernetesNodeNamePrefix {
+				value := aws.StringValue(tag.Value)
+				if len(value) > 0 {
+					return types.NodeName(aws.StringValue(tag.Value))
+				}
+				glog.Infof("Tag %s was found for instance %s, but had no value", TagNameKubernetesNodeNamePrefix, aws.StringValue(i.InstanceId))
+			}
+		}
+		glog.Infof("Tag %s not found for instance %s", TagNameKubernetesNodeNamePrefix, aws.StringValue(i.InstanceId))
+	}
 	return types.NodeName(aws.StringValue(i.PrivateDnsName))
```

Comment:
I assume a custom tag value is mutable on an instance. If so, changing the value of the custom tag after a node has been added to the cluster will cause problems... docs should make that clear

Reply:
Can you point me to where the docs are?

Reply:
I meant the docs to be added around this feature :)